### PR TITLE
Update certs-from-mozilla.py

### PR DESCRIPTION
**Pull request by gnorbsl**
Check if ar exists, if not tell the user what to get to prevent issue #7300 also dynamically get certs instead of hardcoded row item https://github.com/esp8266/Arduino/pull/7573#issuecomment-686192353

### CODE REVIEW

**Comment by d-a-v** [comment]
Thanks !
Our `ar` does not come from espressif but rather from the script `tools/get.py` which reads its sources from [`package/package_esp8266com_index.template.json`](https://github.com/esp8266/Arduino/blob/master/package/package_esp8266com_index.template.json).

**Comment by gnorbsl** [comment]
@d-a-v So assuming someone is a new user that just wants to make SSL requests with this. How does he find ar or get.py?

Is there any documentation that I can link to that shows the user how to get ar?

**Comment by d-a-v** [comment]
> Is there any documentation that I can link to that shows the user how to get ar?

There are three way to install:
- arduino ide: toolchain is automatically downloaded to `tools/xtensa-lx106-elf/`
- git: documentation asks user to run `cd tools; python get.py`. It will download the right toolchain to the same place
- platformio: we don't document it (pio has it own documentation)

> I assume ar is /Java/hardware/tools/avr/bin/avr-ar?

You are running Arduino IDE on macOS. That one is not the good one.
You should have: `Java/hardware/esp8266com/esp8266/tools/xtensa-lx106-elf/xtensa-lx106-elf/bin/ar` and `Java/hardware/esp8266com/esp8266/tools/xtensa-lx106-elf/bin/xtensa-lx106-elf-ar`.
That is the automatic installation mentioned above.

**Comment by gnorbsl** [comment]
@d-a-v Thank you,

can I just link your comment in the python script so users that stumble over this issue can see what to do?

**Comment by d-a-v** [comment]
Not this comment's URL. That would be sufficient :
```
(esp8266-arduino-core-root-path)/tools/xtensa-lx106-elf/bin/xtensa-lx106-elf-ar
```

**Comment by d-a-v** [comment]
You do have it.
Is it not `/Application/Arduino/whatever/Java/hardware/esp8266com/esp8266/tools/xtensa-lx106-elf/bin/xtensa-lx106-elf-ar` ?

**Comment by d-a-v** [comment]
@earlephilhower Do we really need xtensa's ar ? Any version of `ar` seems to be sufficient.

Anyway, we know for sure that we always have this one `(esp8266-arduino-core)/hardware/esp8266com/esp8266/tools/xtensa-lx106-elf/xtensa-lx106-elf/bin/ar` because it is shipped with the necessary toolchain.
Other versions may work but are not be always available on every setup (AVR's `ar` not with PIO, linux's `ar` now with bare windows)

**Comment by gnorbsl** [comment]
I don't have Arduino in Application since I never moved it there.  All I have in the Arduino package contents is a folder named arduino and tools, both have an avr folder inside.

"we" excludes myself then I guess. If others have the path then thats fine. I just got ar from somewhere else and ran the script inside the folder with ar and that fixed it.

**Comment by JiriBilek** [comment]
I suggest you to test the existence of openssl as well.

**Comment by d-a-v** [comment]
I think we should update the path with this one because it is always present in a working setup (the current one is `.../xtensa-lx106-elf-ar`, but the script uses `ar`) :
```
(esp8266-arduino-core)/hardware/esp8266com/esp8266/tools/xtensa-lx106-elf/xtensa-lx106-elf/bin/ar
```

@JiriBilek that's right

```
if which('ar') is None and not os.path.isfile('./ar') and not os.path.isfile('./ar.exe'):
    raise Exception("You need the program 'ar' from xtensa-lx106-elf found here: (esp8266-arduino-core)/hardware/esp8266com/esp8266/tools/xtensa-lx106-elf/xtensa-lx106-elf/bin/ar")
if which('openssl') is None and not os.path.isfile('./openssl') and not os.path.isfile('./openssl.exe'):
    raise Exception("You need to have openssl in PATH, installable from https://www.openssl.org/")
```

**Comment by gnorbsl** [comment]
changed path and added openssl check @d-a-v

File: libraries/ESP8266WiFi/examples/BearSSL_CertStore/certs-from-mozilla.py [hunk 1]
```diff
@@ -20,8 +22,11 @@
     from StringIO import StringIO
 except Exception:
     from io import StringIO
-
-# Mozilla's URL for the CSV file with included PEM certs
+#check if ar is available
+if which('ar') is None and not os.path.isfile('./ar') and not os.path.isfile('./ar.exe'):
+    raise Exception("You need the program 'ar' from xtensa-lx106-elf found here: (esp8266-arduino-core)/hardware/esp8266com/esp8266/tools/xtensa-lx106-elf/xtensa-lx106-elf/bin/ar")
+if which('openssl') is None and not os.path.isfile('./openssl') and not os.path.isfile('./openssl.exe'):
+    raise Exception("You need to have openssl in PATH, installable from https://www.openssl.org/")# Mozilla's URL for the CSV file with included PEM certs
```
Comment (d-a-v) [inline libraries/ESP8266WiFi/examples/BearSSL_CertStore/certs-from-mozilla.py:29]: Maybe a line break/feed for the comment ?

Comment (gnorbsl) [reply libraries/ESP8266WiFi/examples/BearSSL_CertStore/certs-from-mozilla.py]: fixed?

**Review by d-a-v — APPROVED** [review]

**Comment by d-a-v** [comment]
Thanks for making this script more generic !

**Pull request merged**